source/platform/include/Gwork/Renderers/Irrlicht.h

```diff
@@ -0,0 +1,74 @@
+/*
+*  Gwork
+*  Copyright (c) 2011 Facepunch Studios
+*  Copyright (c) 2013-17 Nick Trout
+*  See license in Gwork.h
+*/
+
+#pragma once
+
+#ifndef GWEN_RENDERERS_IRRLICHT_H
+#define GWEN_RENDERERS_IRRLICHT_H
+
+#include "gwork\BaseRender.h"
+#include <Irrlicht.h>
+
+namespace Gwk
+{
+    namespace Renderer
+    {
+        //
+        //  Irrlicht ICacheToTexture
+        class IrrlichtCTT;
+
+        class IrrlichtResourceLoader : public ResourceLoader
+        {
+            ResourcePaths& m_paths;
+            irr::video::IVideoDriver* Driver;
+        public:
+            IrrlichtResourceLoader(irr::video::IVideoDriver* VideoDriver, ResourcePaths& paths) : Driver(VideoDriver), m_paths(paths) {}
+
+            Font::Status LoadFont(Font& font) override;
+            void FreeFont(Font& font) override;
+
+            Texture::Status LoadTexture(Texture& texture) override;
+            void FreeTexture(Texture& texture) override;
+        };
+
+        //
+        //  Irrlicht Renderer
+        class GWK_EXPORT Irrlicht : public Gwk::Renderer::Base
+        {
+            IrrlichtCTT* m_CTT;
+            irr::video::IVideoDriver* Driver;
+            irr::video::SColor DrawColor;
+            irr::core::rect<irr::s32> ClipRect;
+            irr::gui::IGUIFont* Text;
+
+        public:
+            Irrlicht(irr::IrrlichtDevice* Device, IrrlichtResourceLoader& loader);
+            ~Irrlicht();
+
+            void SetDrawColor(Gwk::Color color);
+
+            void StartClip();
+            void EndClip();
+
+            Gwk::Color PixelColor(Gwk::Texture* pTexture, unsigned int x, unsigned int y, const Gwk::Color & col_default = Gwk::Color(255, 255, 255, 255));
```

Comment:
If you put "override" at the end of each of these declarations it will give you an error if it doesn't match the base definition. This would have caught the "colour" renaming problem. 🎩

Reply:
To be honest i'm not sure what "override" does, I saw the other renderers do it so I followed suit.